src/main/java/org/wildfly/security/http/HttpAuthenticator.java

```diff
@@ -120,39 +121,43 @@ private SecurityIdentity login(String username, String password, String mechanis
             return null;
         }
 
-        ServerAuthenticationContext authenticationContext;
-        if(WildFlySecurityManager.isChecking()) {
-            authenticationContext = AccessController.doPrivileged((PrivilegedAction<ServerAuthenticationContext>) () -> securityDomain.createNewAuthenticationContext());
-        } else {
-            authenticationContext = securityDomain.createNewAuthenticationContext();
-        }
-
-        final PasswordGuessEvidence evidence = new PasswordGuessEvidence(password.toCharArray());
+        PasswordGuessEvidence evidence = null;
+        ServerAuthenticationContext authenticationContext = null;
```

Comment:
This is not an improvement IMO.  A cleaner solution would be to use a try-with-resources for the ServerAuthenticationContext and leave the original try/finally block alone.  Having variables which start as null and change state later on means it's more work for the compiler static analysis, which in turn means it's more work for the human static analysis as well.  Using a final variable with try/finally or try-with-resources means that the state can never change, which makes state-change related bugs impossible.

Reply:
Doing it this way also means a smaller diff (ignoring whitespace).